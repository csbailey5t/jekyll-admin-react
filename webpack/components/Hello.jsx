import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setCurrentTime } from '../actionCreators';

const Hello = props => (
  <div>
    <h1>{props.currentTime}</h1>
    <input
      onChange={props.handleTimeChange}
      value={props.currentTime}
      type="text"
      placeholder="time"
    />
  </div>
);

Hello.propTypes = {
  currentTime: PropTypes.string,
  handleTimeChange: PropTypes.func
};

Hello.defaultProps = {
  currentTime: '',
  handleTimeChange: () => {}
};

const mapStateToProps = state => ({ currentTime: state.currentTime });
const mapDispatchToProps = dispatch => ({
  handleTimeChange(event) {
    dispatch(setCurrentTime(event.target.value));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Hello);
