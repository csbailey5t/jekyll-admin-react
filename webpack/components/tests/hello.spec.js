import React from 'react';
import Hello from '../Hello';
import renderer from 'react-test-renderer';

test('Hello component renders as expected', () => {
  const component = renderer.create(<Hello />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
