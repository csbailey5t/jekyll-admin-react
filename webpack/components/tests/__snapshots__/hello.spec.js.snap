// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders as expected 1`] = `
<h1>
  Hello!
</h1>
`;
